app.py

```diff
@@ -1,74 +1,127 @@
 #!/usr/bin/env python3
 from flask import *
 from requests import post
-from json import load, dump
 from re import sub
 
 with open("config.json") as f:
     config = json.load(f)
 
+assert "secret" in config, "Gitlab secret token in config not defined"
+assert "token" in config, "Discord token in config not defined"
+
+if "listen" not in config:
+    config["listen"] = "0.0.0.0"
+if "port" not in config:
+    config["port"] = 25431
 app = Flask(__name__)
 
+
 def handle_push(body):
     """ Handle GitLab push event webhook
-        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md#push-events """
+        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/user/project/integrations/webhooks.md#push-events """
     tcc = body["total_commits_count"]
     commitfmt = "`{id:.7}` **{author[name]}**: {message}"
     body.update({"branch": body["ref"].split("/")[-1],
                  "number": tcc if tcc > 1 else "a",
                  "cmt_plural": "s" if tcc != 1 else "",
                  "commits": "\n".join(commitfmt.format(**c) for c in body["commits"])
-                })
+                 })
     return """:round_pushpin: <{project[web_url]}/commits/{branch}>
 **{user_name}** pushed {number} commit{cmt_plural}:
 {commits}""".format(**body)
 
+
 def handle_tag(body):
     """ Handle GitLab tag push event webhook
-        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md#tag-events """
+        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/user/project/integrations/webhooks.md#tag-events """
     body.update({"tag": body["ref"].split("/")[-1]})
     return """:label: <{project[web_url]}/tree/{tag}>
 **{user_name}** pushed a tag: {tag}""".format(**body)
 
+
 def handle_issue(body):
     """ Handle GitLab issue event webhook
-        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md#issues-events """
-    a = body["object_attributes"]["action"]
+        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/user/project/integrations/webhooks.md#issues-events """
+    try:
+        a = body["object_attributes"]["action"]
+    except KeyError:
+        a = "Touch"
```

Comment:
Touch?

Reply:
Did not know what to put there, as that is not supposed to happen but does happen when I try to test the hook, so it resultes in <Name> touched  the  issue.